### PR TITLE
Nebula first aid pouches

### PR DESCRIPTION
**Pull request by TheLordME**
## About The Pull Request
grafik

## Why It's Good For The Game
Kinda cool, plus you can drop them around places where a medkit would be overkill

## Changelog
:cl:
add: Added first aid pouches you might know from bay or polaris
code: Added a single use storage subtype
refactor: MREs are now single_use subtypes
/:cl:

### CODE REVIEW

**Comment by TheLordME** [comment]
zandario is hogging the maps so I cant map them in :(

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+	if(length(contents))
+		make_exact_fit()
+	update_icon()
+
+/obj/item/storage/single_use/med_pouch/update_icon()
+	. = ..()
+	if(!cross_overlay)
+		cross_overlay = overlay_image(icon, "cross", flags = RESET_COLOR)
+	add_overlay(cross_overlay)
+	icon_state = "pack[opened]"
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:41]: Split up appearance updates
```suggestion
	update_appearance()

/obj/item/storage/single_use/med_pouch/update_icon_state()
	. = ..()
	icon_state = "pack[opened]"

/obj/item/storage/single_use/med_pouch/update_overlays()
	. = ..()
	if(!cross_overlay)
		cross_overlay = overlay_image(icon, "cross", flags = RESET_COLOR)
	add_overlay(cross_overlay)

```

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+/obj/item/storage/single_use/med_pouch/examine(mob/user)
+	. = ..()
+	. += instructions
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:45]: Okay yeah we can totally just blockquote this

File: code/game/objects/random/misc.dm [hunk 3]
```diff
@@ -591,16 +591,16 @@
...
+	return pick(/obj/item/storage/single_use/mre,
+				/obj/item/storage/single_use/mre/menu2,
+				/obj/item/storage/single_use/mre/menu3,
+				/obj/item/storage/single_use/mre/menu4,
+				/obj/item/storage/single_use/mre/menu5,
+				/obj/item/storage/single_use/mre/menu6,
+				/obj/item/storage/single_use/mre/menu7,
+				/obj/item/storage/single_use/mre/menu8,
+				/obj/item/storage/single_use/mre/menu9,
+				/obj/item/storage/single_use/mre/menu10)
```
Comment (Zandario) [inline code/game/objects/random/misc.dm:603]:
```suggestion
	return pick(
		/obj/item/storage/single_use/mre,
		/obj/item/storage/single_use/mre/menu2,
		/obj/item/storage/single_use/mre/menu3,
		/obj/item/storage/single_use/mre/menu4,
		/obj/item/storage/single_use/mre/menu5,
		/obj/item/storage/single_use/mre/menu6,
		/obj/item/storage/single_use/mre/menu7,
		/obj/item/storage/single_use/mre/menu8,
		/obj/item/storage/single_use/mre/menu9,
		/obj/item/storage/single_use/mre/menu10,
	)
```

File: code/game/objects/random/misc.dm [hunk 4]
```diff
@@ -610,14 +610,14 @@
...
-				/obj/item/storage/mrebag/menu7,
-				/obj/item/storage/mrebag/menu8)
+	return pick(/obj/item/storage/single_use/mrebag,
+				/obj/item/storage/single_use/mrebag/menu2,
+				/obj/item/storage/single_use/mrebag/menu3,
+				/obj/item/storage/single_use/mrebag/menu4,
+				/obj/item/storage/single_use/mrebag/menu5,
+				/obj/item/storage/single_use/mrebag/menu6,
+				/obj/item/storage/single_use/mrebag/menu7,
+				/obj/item/storage/single_use/mrebag/menu8)
```
Comment (Zandario) [inline code/game/objects/random/misc.dm:620]:
```suggestion
	return pick(
		/obj/item/storage/single_use/mrebag,
		/obj/item/storage/single_use/mrebag/menu2,
		/obj/item/storage/single_use/mrebag/menu3,
		/obj/item/storage/single_use/mrebag/menu4,
		/obj/item/storage/single_use/mrebag/menu5,
		/obj/item/storage/single_use/mrebag/menu6,
		/obj/item/storage/single_use/mrebag/menu7,
		/obj/item/storage/single_use/mrebag/menu8,
	)
```

File: code/modules/cargo/supplypacks/misc.dm [hunk 5]
```diff
@@ -148,24 +148,24 @@
...
+	contains = list(/obj/item/storage/single_use/mre,
+					/obj/item/storage/single_use/mre/menu2,
+					/obj/item/storage/single_use/mre/menu3,
+					/obj/item/storage/single_use/mre/menu4,
+					/obj/item/storage/single_use/mre/menu5,
+					/obj/item/storage/single_use/mre/menu6,
+					/obj/item/storage/single_use/mre/menu7,
+					/obj/item/storage/single_use/mre/menu8,
+					/obj/item/storage/single_use/mre/menu9,
+					/obj/item/storage/single_use/mre/menu10)
```
Comment (Zandario) [inline code/modules/cargo/supplypacks/misc.dm:160]:
```suggestion
	contains = list(
		/obj/item/storage/single_use/mre,
		/obj/item/storage/single_use/mre/menu2,
		/obj/item/storage/single_use/mre/menu3,
		/obj/item/storage/single_use/mre/menu4,
		/obj/item/storage/single_use/mre/menu5,
		/obj/item/storage/single_use/mre/menu6,
		/obj/item/storage/single_use/mre/menu7,
		/obj/item/storage/single_use/mre/menu8,
		/obj/item/storage/single_use/mre/menu9,
		/obj/item/storage/single_use/mre/menu10,
	)
```

File: code/modules/cargo/supplypacks/misc.dm [hunk 6]
```diff
@@ -148,24 +148,24 @@
...
 	cost = 50
 	container_type = /obj/structure/closet/crate/freezer
 	container_name = "ready to eat rations"
 
 /datum/supply_pack/misc/paste_rations
 	name = "Emergency - Paste"
 	contains = list(
-			/obj/item/storage/mre/menu11 = 2
+			/obj/item/storage/single_use/mre/menu11 = 2
 			)
```
Comment (Zandario) [inline code/modules/cargo/supplypacks/misc.dm:169]:
```suggestion
		/obj/item/storage/single_use/mre/menu11 = 2,
	)
```

File: code/modules/cargo/supplypacks/misc.dm [hunk 7]
```diff
@@ -174,7 +174,7 @@
 /datum/supply_pack/misc/medical_rations
 	name = "Emergency - VitaPaste"
 	contains = list(
-			/obj/item/storage/mre/menu13 = 2
+			/obj/item/storage/single_use/mre/menu13 = 2
 			)
```
Comment (Zandario) [inline code/modules/cargo/supplypacks/misc.dm:178]:
```suggestion
		/obj/item/storage/single_use/mre/menu13 = 2,
	)
```

**Review by Zandario — COMMENTED** [review]
Babe is 6AM, it's time for your Review Bomb.

<details>
<summary>tl;dr</summary>
None of my notes are block worthy, and are moreso focused on minor standardizations.

- After this PR we should probably deal with standardizing `ITEM_SIZE` naming.
- Do heavily recommend doing my update_icon() split.
- Same with my blockquote suggestion. It might be a good idea to see if we can remove a lot of the redundant strings for the instructions, but it's heavily out of scope.

</details>

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+	\t6) Contact the medical team with your location.
+	7) Stay in place once they respond.\
+		"}
+	starts_with = list(
+		/obj/item/reagent_containers/hypospray/autoinjector,
+		/obj/item/reagent_containers/hypospray/autoinjector/pouch_pain,
+		/obj/item/reagent_containers/pill/inaprovaline/pouch,
+		/obj/item/reagent_containers/pill/bicaridine/pouch,
+		/obj/item/stack/medical/bruise_pack,
+		)
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:76]:
```suggestion
		/obj/item/stack/medical/bruise_pack,
	)
```

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+	\t6) Contact the medical team with your location.
+	7) Stay in place once they respond.\
+		"}
+	starts_with = list(
+		//obj/item/chems/hypospray/autoinjector/pouch_auto/nanoblood,//maybe we add nanoblood one day
+		/obj/item/reagent_containers/hypospray/autoinjector/pouch_pain,
+		/obj/item/reagent_containers/hypospray/autoinjector/burn_pouch,//used to be an adrenaline one, but adrenaline is kinda bad here
+		/obj/item/reagent_containers/pill/kelotane,
+		/obj/item/reagent_containers/pill/spaceacillin,
+		/obj/item/stack/medical/ointment
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:98]: Minor
```suggestion
		// /obj/item/chems/hypospray/autoinjector/pouch_auto/nanoblood, // Maybe we add nanoblood one day.
		/obj/item/reagent_containers/hypospray/autoinjector/burn_pouch, // Used to be an adrenaline one, but adrenaline is kinda bad here.
		/obj/item/reagent_containers/hypospray/autoinjector/pouch_pain,
		/obj/item/reagent_containers/pill/kelotane,
		/obj/item/reagent_containers/pill/spaceacillin,
		/obj/item/stack/medical/ointment,
```

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+	\t7) Update the medical team with your new location.\n\
+	8) Stay in place once they respond.\
+		"}
+	starts_with = list(
+		/obj/item/reagent_containers/hypospray/autoinjector,
+		/obj/item/reagent_containers/hypospray/autoinjector/pouch_oxy,//used to be an inhaler
+		//obj/item/chems/hypospray/autoinjector/pouch_auto/adrenaline,
+		/obj/item/reagent_containers/pill/inaprovaline/pouch,
+		/obj/item/reagent_containers/pill/dexalin = 2,
+	)
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:121]: Ditto
```suggestion
	starts_with = list(
		// /obj/item/chems/hypospray/autoinjector/pouch_auto/adrenaline,
		/obj/item/reagent_containers/hypospray/autoinjector,
		/obj/item/reagent_containers/hypospray/autoinjector/pouch_oxy, // Used to be an inhaler.
		/obj/item/reagent_containers/pill/dexalin = 2,
		/obj/item/reagent_containers/pill/inaprovaline/pouch,
	)
```

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 11]
```diff
@@ -0,0 +1,327 @@
...
+	\t3) Apply all autoinjectors to the injured party.\n\
+	\t4) Force the injured party to swallow all pills.\n\
+	\t5) Contact the medical team with your location.\n\
+	\t6) Stay in place once they respond.\
+		"}
+	starts_with = list(
+		/obj/item/reagent_containers/hypospray/autoinjector/detox = 2,
+		/obj/item/reagent_containers/hypospray/autoinjector/pouch_tox,
+		/obj/item/reagent_containers/pill/antitox = 2
+		)
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:139]: Ditto
```suggestion
	starts_with = list(
		/obj/item/reagent_containers/hypospray/autoinjector/detox = 2,
		/obj/item/reagent_containers/hypospray/autoinjector/pouch_tox,
		/obj/item/reagent_containers/pill/antitox = 2,
	)
```

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 12]
```diff
@@ -0,0 +1,327 @@
...
+	\t3) Apply all autoinjectors to the injured party.\n\
+	\t4) Force the injured party to swallow all pills.\n\
+	\t5) Contact the medical team with your location.
+	6) Stay in place once they respond.\
+		"}
+	starts_with = list(
+		/obj/item/reagent_containers/hypospray/autoinjector/detox = 1,
+		/obj/item/reagent_containers/hypospray/autoinjector/pouch_rad = 2,
+		/obj/item/reagent_containers/pill/antitox = 2
+	)
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:157]: Ditto
```suggestion
	starts_with = list(
		/obj/item/reagent_containers/hypospray/autoinjector/detox = 1,
		/obj/item/reagent_containers/hypospray/autoinjector/pouch_rad = 2,
		/obj/item/reagent_containers/pill/antitox = 2,
	)
```

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 13]
```diff
@@ -0,0 +1,327 @@
...
+	8) Stay in place once they respond.\
+		"}
+	starts_with = list(
+		/obj/item/reagent_containers/hypospray/autoinjector,
+		/obj/item/reagent_containers/hypospray/autoinjector/pouch_oxy,//used to be an inhaler,
+		/obj/item/reagent_containers/hypospray/autoinjector/detox,
+		/obj/item/reagent_containers/pill/carbon,
+		/obj/item/reagent_containers/hypospray/autoinjector/pouch_antialcohol
+		//obj/item/chems/hypospray/autoinjector/pouch_auto/adrenaline
+	)
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:179]:
```suggestion
	starts_with = list(
		// /obj/item/chems/hypospray/autoinjector/pouch_auto/adrenaline,
		/obj/item/reagent_containers/hypospray/autoinjector,
		/obj/item/reagent_containers/hypospray/autoinjector/detox,
		/obj/item/reagent_containers/hypospray/autoinjector/pouch_antialcohol,
		/obj/item/reagent_containers/hypospray/autoinjector/pouch_oxy, // Used to be an inhaler.
		/obj/item/reagent_containers/pill/carbon,
	)
```

File: code/game/objects/items/storage/single_use/med_pouch.dm [hunk 14]
```diff
@@ -0,0 +1,327 @@
+/*
+Single Use Emergency Pouches
+ */
+
+/obj/item/storage/single_use/med_pouch
```
Comment (Zandario) [inline code/game/objects/items/storage/single_use/med_pouch.dm:5]: DMDoc my beloved.
```suggestion
/**
 * Single Use Emergency Pouches
 */
/obj/item/storage/single_use/med_pouch
```

**Review by Zandario — COMMENTED** [review]
So I forgot some things...

File: code/__DEFINES/inventory_storage.dm [hunk 15]
```diff
@@ -6,6 +6,16 @@
...
+#define ITEM_SIZE_TINY           1
+#define ITEM_SIZE_SMALL          2
+#define ITEM_SIZE_NORMAL         3
+#define ITEM_SIZE_LARGE          4
+#define ITEM_SIZE_HUGE           5
+#define ITEM_SIZE_GARGANTUAN     6
+#define ITEM_SIZE_NO_CONTAINER   10 // Use this to forbid item from being placed in a container.
+#define ITEM_SIZE_STRUCTURE      20
+
 /// Use this to forbid item from being placed in a container.
```
[the remark this answers is not in the document]
Comment (TheLordME) [reply code/__DEFINES/inventory_storage.dm:19]: I am aware that its a duplication of an existing define, but everytime I try to port something from nebula or bay I wonder "whats our equivalent again?" and defines shouldnt be a major performace impact anyways

**Review by TheLordME — COMMENTED** [review]
Its my PR and I think I did a pretty good job

**Pull request merged**